catroid/src/main/java/org/catrobat/catroid/merge/MergeProjects.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.merge;
+
+import android.content.Context;
+
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Scene;
+import org.catrobat.catroid.content.XmlHeader;
+import org.catrobat.catroid.io.XstreamSerializer;
+import org.catrobat.catroid.utils.Utils;
+
+public class MergeProjects {
+	private Project firstProject = null;
+	private Project secondProject = null;
+	private Project mergedProject = null;
+	private ConflictHelper conflictHelper = new ConflictHelper();
+	private Context context;
+
+	public MergeProjects(Project firstProject, Project secondProject, Context context) {
+		this.firstProject = firstProject;
+		this.secondProject = secondProject;
+		this.context = context;
+	}
+
+	public MergeProjects(Project firstProject, Project secondProject) {
+		this.firstProject = firstProject;
+		this.secondProject = secondProject;
+	}
+
+	public Project getMergedProject() {
+		return mergedProject;
+	}
+
+	public boolean performMerge(String mergeProjectName) {
+		if (conflictHelper.conflictExist(firstProject, secondProject)) {
+			return false;
+		}
+		mergedProject = new Project(context, mergeProjectName);
+
+		if (mergedProject.getDefaultScene() != null) {
+			mergedProject.removeScene(mergedProject.getDefaultScene());
+		}
+
+		createProjectHeader();
+
+		addResources();
+		XstreamSerializer.getInstance().saveProject(mergedProject);
+
+		return true;
+	}
```

Comment:
Cean Code -> Functions should only have one purpose. This is like a super function which checks for Conflicts , creates a project, names the project (even if there's an uncalled dialog), and generates (XML) + saves the project

Reply:
I did that specifically to condense the merge such that it would make usage easier and cleaner. Else I would need to call all the inner functions when I am merging.

Reply:
I think this would be OK if you say it's not good to split up the functions, but you have to be aware of what the inner functions do and keep in mind how to react if e.g. saveProject doesn't worked